### PR TITLE
made duplicates adjacent

### PR DESCRIPTION
**Pull request by apex-omontgomery**
Found a few duplicate terms. Which do you prefer to have?

### CODE REVIEW

File: generator/raw_out.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from collections import defaultdict
+from typing import DefaultDict
+from typing import List
+
+import requests
+
+TERM_URL = 'https://raw.githubusercontent.com/togakangaroo/tech-terms/master/terms.org'
+
+
+def convert_to_dict_list(items: List[dict]) -> DefaultDict[str, list]:
```
Comment (togakangaroo) [inline generator/raw_out.py]: so for my own understanding, any reason not to do this as a dictionary generator?

      return {x['term']: x['definition'] for x in items}

You would then not need to pass in a list either, you could keep it a sequence and wouldn't have to double-load into memory.

File: generator/raw_out.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    return terms
+
+
+def parse_ank_raw_from_source() -> DefaultDict[str, list]:
+    r = requests.get(TERM_URL)
+    content = r.content
+
+    # this regex matches 
+    two_col_org_row = re.compile('^\\s*\\|\\s*(?P<term>.*?)\\s*\\|\\s*(?P<definition>.*?)\\s*\\|\\s*$')
+    lines = content.splitlines()
```
Comment (togakangaroo) [inline generator/raw_out.py]: So this is missing the part of my script where it made sure that there was exactly a term and a definition. If we want to add some header text to the table I wouldn't want it to break this.

Also maybe refactor the existing regex so its not just duplicated everywhere?

Comment (apex-omontgomery) [reply generator/raw_out.py]: Okay, split it out into smaller components.

File: generator/raw_out.py (hunk 1 above)
Comment (apex-omontgomery) [reply generator/raw_out.py]: Sounds good.

**Comment by apex-omontgomery** [comment]
initial time:
image
refactor time:
image

**Comment by apex-omontgomery** [comment]
initial time intensive:
image

refactor time intensive:
image

Looks like the most expensive was creating the dictionary item and the default dict is slightly slower.

**Pull request merged**